src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -888,6 +888,14 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
 
                         if (body != null && ((methodSymbol.ContainingType.IsStructType() && !methodSymbol.IsImplicitConstructor) || instrumentForDynamicAnalysis))
                         {
+                            if (instrumentForDynamicAnalysis && methodSymbol.IsImplicitConstructor)
```

Comment:
>instrumentForDynamicAnalysis && methodSymbol.IsImplicitConstructor [](start = 32, length = 66)

It feels like the condition should be ```!methodSymbol.ContainingType.IsStructType() || methodSymbol.IsImplicitConstructor```.
In general, it looks like the logic here becomes rather hard to follow with code duplication on different levels of branching. Consider if we could separate the decision making about doing additional analysis and merging initializers into body into two independent ```if``` statements.